src/app/pages/home-search/home-search.page.ts

```diff
@@ -1,121 +1,248 @@
   }
-  trimLeft(inputString: string): string {
-    return this.searchText = inputString.replace(/^\s+/, '');
+
+  onPageChange(event){
+    this.page = event.page,
+    this.pageSize = event.pageSize;
+    this.fetchSessionList()
   }
 
-  search(){
-    if(this.searchText!=""){
-      switch(this.type) {
-        case 'all-sessions':
-          this.getSessionsList();
+  async eventAction(event) {
+    this.user = await this.localStorage.getLocalData(localKeys.USER_DETAILS)
+    if (this.user.about) {
+      switch (event.type) {
+        case 'cardSelect':
+          this.router.navigate([`/${CommonRoutes.SESSIONS_DETAILS}/${event.data.id}`]);
           break;
-        case 'mentor-profile':
-          this.getMentorList();
+
+        case 'joinAction':
+          await this.sessionService.joinSession(event.data)
+          this.fetchSessionList()
+          break;
+
+        case 'enrollAction':
+          let enrollResult = await this.sessionService.enrollSession(event.data.id);
+          if (enrollResult.result) {
+            this.toast.showToast(enrollResult.message, "success")
+            this.fetchSessionList()
+          }
+          break;
+
+        case 'startAction':
+          this.sessionService.startSession(event.data.id).then(async () => {
+            var obj = { page: this.page, limit: this.limit, searchText: "" };
+            if(this.profileService.isMentor){
+              this.createdSessions = await this.sessionService.getAllSessionsAPI(obj);
+            }
+          })
           break;
       }
     } else {
-      this.toast.showToast("Please provide a valid text for search","danger")
+      this.profileService.upDateProfilePopup()
     }
   }
 
-  segmentChanged(event){
-    this.results=[]
-    this.type = event.target.value;
-    if(this.searchText !== "")
-    this.search();
+  locationBack(){
+    this.location.back()
   }
 
-  async onSessionAction(event){
-    switch (event.type) {
-      case 'cardSelect':
-        this.router.navigate([`/${CommonRoutes.SESSIONS_DETAILS}/${event.data.id}`])
-        break;
+  extractLabels(data) {
+    this.chips = [];
+    for (const key in data) {
+      if (data.hasOwnProperty(key)) {
+        this.chips.push(...data[key]);
+      }
+    }
+  }
 
-      case 'joinAction':
-        this.sessionService.joinSession(event.data)
-        this.search();
-        break;
+  removeChip(chip: string, index: number) {
+    this.chips.splice(index, 1);
+    this.removeFilteredData(chip);
+    this.getFilteredData();
+    this.fetchSessionList()
+  }
 
-      case 'enrollAction':
-        let enrollResult = await this.sessionService.enrollSession(event.data.id);
-        if(enrollResult.result){
-          this.toast.showToast(enrollResult.message, "success")
-          this.search();
-        }
-        break;
+  closeCriteriaChip(){
+    this.criteriaChip = "";
+    this.criteriaChipName = "";
```

Comment:
why two variables? cant you use it as a single object with name and label if needed?